modules/widgets/top-posts.php

```diff
@@ -105,6 +130,14 @@ function update( $new_instance, $old_instance ) {
 			$instance['count'] = 10;
 		}
 
+		$allowed_post_types = array_values( get_post_types( array( 'public' => true ) ) );
+		$instance['types'] = $new_instance['types'];
+		foreach( $new_instance['types'] as $type ) {
+			if ( ! in_array( $type, $allowed_post_types ) ) {
+				unset( $type );
```

Comment:
This won't actually unset, you need to unset a member of the `$new_isntance['types']` array by key.